jwt.go

```diff
@@ -19,13 +19,193 @@ var TimeFunc = time.Now
 // Header of the token (such as `kid`) to identify which key to use.
 type Keyfunc func(*Token) (interface{}, error)
 
+// For a type to be a Claims object, it must just have a Valid method that determines
+// if the token is invalid for any supported reason
+type Claimer interface {
```

Comment:
The `er` idiomatically mean they have a method that does that (`Reader`, `Closer`, `Stringer`).  There are plenty of use cases where the interface is the most generic form and it's described that way. `hash.Hash` for example, is an interface.